Layouts/Base.inc

```diff
@@ -315,23 +327,44 @@ $endif
 ;**** **** **** **** ****
 ; ADC and temperature measurement
 ;**** **** **** **** ****
-TEMP_LIMIT		EQU	49			; Temperature measurement ADC value for which main motor power is limited at 80degC (low byte, assuming high byte is 1)
-TEMP_LIMIT_STEP	EQU	9			; Temperature measurement ADC value increment for another 10degC
+; Temperature measurement ADC value for which main motor power is limited at 80degC (low byte, assuming high byte is 1)
+IF MCU_48MHZ < 2
+	TEMP_LIMIT		EQU	49
+ELSEIF MCU_48MHZ == 2
+	TEMP_LIMIT		EQU	92
+ENDIF
+TEMP_LIMIT_STEP		EQU	9		; Temperature measurement ADC value increment for another 10degC
 
 Initialize_Adc MACRO
-	mov	REF0CN, #0Ch				;; Set vdd (3.3V) as reference. Enable temp sensor and bias
-IF MCU_48MHZ == 0
-	mov	ADC0CF, #59h				;; ADC clock 2MHz, PGA gain 1
-ELSE
-	mov	ADC0CF, #0B9h				;; ADC clock 2MHz, PGA gain 1
-ENDIF
+	IF MCU_48MHZ < 2
+		mov	REF0CN, #0Ch			;; Set vdd (3.3V) as reference. Enable temp sensor and bias
+		IF MCU_48MHZ == 0
+			mov	ADC0CF, #59h		;; ADC clock 2MHz, PGA gain 1
+		ELSEIF MCU_48MHZ == 1
+			mov	ADC0CF, #0B9h		;; ADC clock 2MHz, PGA gain 1
+		ENDIF
+	ELSEIF MCU_48MHZ == 2
+		mov	CLKGRP0, #18h			;; Enable SAR clock, at 12MHz
+		mov	ADC0CF1, #0Ah			;; 800ns tracking time
+		mov	ADC0CF2, #2Fh			;; 1.65V reference
+	ENDIF
+
 	mov	ADC0MX, #10h				;; Select temp sensor input
-	mov	ADC0CN0, #80h				;; ADC enabled
-	mov	ADC0CN1, #01h				;; Common mode buffer enabled
+	IF MCU_48MHZ < 2
+		mov	ADC0CN0, #80h			;; ADC enabled
+		mov	ADC0CN1, #01h			;; Common mode buffer enabled
+	ELSEIF MCU_48MHZ == 2
+		mov	ADC0CN0, #85h			;; ADC enabled, gain 0.5, temp sensor enabled
```

Comment:
The gain control is a general setting for the ADC - so are we sure that changing this is not going to be an issue anywhere else?

Also I was suspicious since the "new" limit was 92 (instead of 98 which would be double of the original TEMP_LIMIT) I though there might be different offsets in regards to the temperature.

Any idea why it would be solved like this in BLHeli_S?

Reply:
@damosvil and i discussed this a bit further on Discord and are not sure if chaning the gain wouldn't impact comparator functionality. For now we agreed to leave it like this and double check this once EDT is ready to go.

Reply:
I reopened this. @damosvil double checked the manual and thinks that changing this will purely affect, the temp readings. Thus I will adapt it in this PR if it does not have any other side effects anyway.

Reply:
OK, now that I could test it, the gain makes no difference, it always shows double the temperature of what it should show, so double the temp limit makes sense in this context.